### PR TITLE
use enum instead of int

### PR DESCRIPTION
**Pull request by karussell**
@easbar as discussed in #1950 :)

### CODE REVIEW

File: core/src/main/java/com/graphhopper/storage/BaseGraph.java [hunk 1]
```diff
@@ -968,14 +968,14 @@ private PointList fetchWayGeometry_(long edgePointer, boolean reverse, int mode,
...
 
-        PointList pillarNodes = new PointList(count + mode, nodeAccess.is3D());
+        PointList pillarNodes = new PointList(FetchWayGeometry.count(count, mode), nodeAccess.is3D());
         if (reverse) {
-            if ((mode & 2) != 0)
+            if (mode == FetchWayGeometry.ALL || mode == FetchWayGeometry.PILLAR_AND_ADJ)
                 pillarNodes.add(nodeAccess, adjNode);
-        } else if ((mode & 1) != 0)
+        } else if (mode == FetchWayGeometry.ALL || mode == FetchWayGeometry.BASE_AND_PILLAR)
             pillarNodes.add(nodeAccess, baseNode);
```
Comment (karussell) [inline core/src/main/java/com/graphhopper/storage/BaseGraph.java]: This is a bit more lengthy now ;)

File: core/src/main/java/com/graphhopper/util/FetchWayGeometry.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package com.graphhopper.util;
+
+public enum FetchWayGeometry {
```
Comment (easbar) [inline core/src/main/java/com/graphhopper/util/FetchWayGeometry.java]: I do not really like the enum name `FetchWayGeometry` too much. Its not a noun and it does not really tell us anything. Maybe we can use this name to communicate that this enum is used to specify how many nodes we fetch? For example `GeometryNodes.ALL`? `FetchMode.ALL`? `WayGeometryFetchMode.ALL`?

File: core/src/main/java/com/graphhopper/storage/BaseGraph.java (hunk 1 above)
Comment (easbar) [reply core/src/main/java/com/graphhopper/storage/BaseGraph.java]: IMO its more lengthy and a lot more readable. With the previous code I did not understand what happened here without looking up what e.g. mode=2 means.

File: core/src/main/java/com/graphhopper/util/FetchWayGeometry.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    TOWER_ONLY, PILLAR_ONLY, BASE_AND_PILLAR, PILLAR_AND_ADJ, ALL;
+
+    public static int count(int count, FetchWayGeometry mode) {
```
Comment (easbar) [inline core/src/main/java/com/graphhopper/util/FetchWayGeometry.java]: The names here could be more expressive. `int count(int count` does not tell me anything about what this method does. Apparently the method returns the total number of nodes that will be returned by `fetchWayGeometry` given the number of pillar nodes? So maybe we can at least rename the `count` parameter to `pillarNodes`? Maybe rename the method to `getFetchedNodeCount` or `countFetchedNodes`?

File: core/src/main/java/com/graphhopper/util/FetchWayGeometry.java (hunk 2 above)
Comment (karussell) [reply core/src/main/java/com/graphhopper/util/FetchWayGeometry.java]: Yes, FetchMode sounds already good.

File: core/src/main/java/com/graphhopper/util/FetchWayGeometry.java (hunk 3 above)
Comment (karussell) [reply core/src/main/java/com/graphhopper/util/FetchWayGeometry.java]: I'll probably move the method into the only class where it is being used then we do not need to care much about the name for now :) (?)

File: core/src/main/java/com/graphhopper/storage/BaseGraph.java [hunk 4]
```diff
@@ -993,16 +993,31 @@ private PointList fetchWayGeometry_(long edgePointer, boolean reverse, int mode,
...
 
             pillarNodes.reverse();
-        } else if ((mode & 2) != 0)
+        } else if (mode == FetchMode.ALL || mode == FetchMode.PILLAR_AND_ADJ)
             pillarNodes.add(nodeAccess, adjNode);
 
         return pillarNodes;
     }
 
+    static int getPointListLength(int count, FetchMode mode) {
```
Comment (easbar) [inline core/src/main/java/com/graphhopper/storage/BaseGraph.java]: I'd still change `count` to `pillarNodes` (if thats what it is?)

**Pull request merged**